git-workflow.md

```diff
@@ -0,0 +1,101 @@
+# Pacific Development `git` workflow
+
+## Single source of truth
+
+The `main` branch of [this](https://github.com/PacificDevelopment/JobSite) remote repository will serve as our **single source of truth**. It will represent the latest version of production-ready code.
+
+## Branch Protection
+`main` branch has special protection rules.
+1. To merge code into `main` you must create a pull request.
+2. The pull request must be approved by at least 1 other person.
+3. No one can push directly to main except the team captains.
+4. In the future Continous integration will also place a status check preventing anyone from pushing directly to main.
+
+
+## Cloning the repository
+
+Your first step will be cloning our remote repository. Navigate to the appropriate folder on your local machine and execute the following command:
+
+```
+git clone https://github.com/PacificDevelopment/JobSite.git <desired_folder_name>
+```
+
+Then change into your newly created directory:
+
+```
+cd <desired_folder_name>
+```
+
+Confirm that you are configured with the correct remote repository:
+
+```
+git remote -v
+```
+
+You should see the following:
+
+```
+origin	https://github.com/PacificDevelopment/JobSite.git (fetch)
+origin	https://github.com/PacificDevelopment/JobSite.git (push)
+```
+
+## Working on a feature
+
+> No work should be done on the `main` branch.
+
+All work should be done on a **feature branch**; that is, an existing or newly created branch designated to the feature on which you are working.
```

Comment:
Could we add a line about what to do if a developer accidentally makes a commit to their local main branch?

Reply:
Added instructions for what to do for 2 scenarios:
1) I accidentally make commits to main that were intended for an existing feature-branch.
2) I accidentally make commits to main that were intended for a new feature-branch.